### PR TITLE
feat: Search part 1

### PR DESCRIPTION
**Pull request by rajsam003**
This PR resolves [MOPLAT-422]

https://user-images.githubusercontent.com/17421923/197531103-8d816b50-c5f5-4828-a8b5-f3d390174c7e.mov

### PR Checklist

- [x] I tested my changes on **iOS** and **Android**.
- [x] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.

### To the reviewers 👀

- [x] I would like **at least one** of the reviewers to run this PR on the simulator or device.

Need help with something? Have a look at our [docs], or get in touch with us.

[docs]: /docs/README.md

[MOPLAT-422]: https://artsyproduct.atlassian.net/browse/MOPLAT-422?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/palette/elements/Pill/Pill.tsx [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+    case "sm":
+      return {
+        height: 50,
+        paddingRight: 20,
+        paddingLeft: 10,
+      }
+  }
+}
+
+export const Pill: React.FC<PillProps> = ({
```
Comment (rajsam003) [inline src/palette/elements/Pill/Pill.tsx:69]: @pvinis Do we have <Pill /> in artsy/palette-token ?

File: src/app/screens/HomeTabs/Search/SearchResult.tsx [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+import { ActivityIndicator } from "react-native"
+import { isTablet } from "react-native-device-info"
+import { graphql, useLazyLoadQuery } from "react-relay"
+import { SearchResultQuery } from "__generated__/SearchResultQuery.graphql"
+import { NavigationScreens } from "app/navigation/Main"
+import { GlobalStore } from "app/store/GlobalStore"
+import { imageSize } from "app/utils/imageSize"
+import { extractNodes } from "shared/utils"
+
+export const SearchResult = ({ searchInput }: { searchInput: string }) => {
```
Comment (rajsam003) [inline src/app/screens/HomeTabs/Search/SearchResult.tsx:13]: Had to add this, as SuspenseWrapper was not helping. It was loading the whole screen.

File: ios/Energy.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -623,7 +623,7 @@
 				COPY_PHASE_STRIP = NO;
 				ENABLE_STRICT_OBJC_MSGSEND = YES;
 				ENABLE_TESTABILITY = YES;
-				"EXCLUDED_ARCHS[sdk=iphonesimulator*]" = "";
+				"EXCLUDED_ARCHS[sdk=iphonesimulator*]" = "arm64 ";
```
[the remark this answers is not in the document]
Comment (pvinis) [reply ios/Energy.xcodeproj/project.pbxproj:626]: it's not

File: src/palette/elements/Pill/Pill.tsx (hunk 1 above)
Comment (pvinis) [reply src/palette/elements/Pill/Pill.tsx:69]: it's not yet. you did good by adding it here. we can move it to palette-mobile soon

File: src/app/screens/HomeTabs/Search/Search.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    []
+  )
+
+  const handleChangeText = (text: string) => {
+    setInputText(text)
+    handleSearch(text)
+  }
+
+  return (
+    <Flex flex={1} pt={insets.top} px={2} backgroundColor="background">
```
Comment (pvinis) [inline src/app/screens/HomeTabs/Search/Search.tsx:54]: no more of that ;)

can you use Screen?

File: src/app/screens/HomeTabs/Search/SearchResult.tsx (hunk 2 above)
Comment (pvinis) [reply src/app/screens/HomeTabs/Search/SearchResult.tsx:13]: yep. that's what we were discussing the other day about having more local suspense wrappers.

**Review by pvinis — COMMENTED** [review]
nice. I think only the Screen comment needs addressing, then it's ready

**Review by pvinis — CHANGES_REQUESTED** [review]
wrong button before.

File: src/app/screens/HomeTabs/Search/Search.tsx (hunk 4 above)
Comment (rajsam003) [reply src/app/screens/HomeTabs/Search/Search.tsx:54]: OOPS. Oh yess. haha

File: ios/Energy.xcodeproj/project.pbxproj (hunk 3 above)
[the remark this answers is not in the document]
Comment (pvinis) [reply ios/Energy.xcodeproj/project.pbxproj:626]: you can revert this file.

also, here's a PR that should solve this git diff generation. https://github.com/artsy/energy/pull/285

File: src/app/screens/HomeTabs/Search/Search.tsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  )
+
+  const handleChangeText = (text: string) => {
+    setInputText(text)
+    handleSearch(text)
+  }
+
+  return (
+    <Screen>
+      <Screen.RawHeader>
```
Comment (pvinis) [inline src/app/screens/HomeTabs/Search/Search.tsx:54]: heeeeeell yea!! nice, sam!

File: src/app/screens/HomeTabs/Search/Search.tsx [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+            placeholder="Search"
+            enableClearButton
+          />
+          <Touchable onPress={() => navigation.goBack()}>
+            <Text variant="xs">Cancel</Text>
+          </Touchable>
+        </Flex>
+      </Screen.RawHeader>
+      <Screen.Body>
+        <Separator />
```
Comment (pvinis) [inline src/app/screens/HomeTabs/Search/Search.tsx:72]: i would move this to the header I think.

**Review by pvinis — APPROVED** [review]
nice. one tiny comment and then feel free to merge, ill assign it back.

File: src/app/screens/HomeTabs/Search/Search.tsx (hunk 6 above)
Comment (rajsam003) [reply src/app/screens/HomeTabs/Search/Search.tsx:72]: Then we would need to wrap <Separator> and this existing header into <Flex> and give mx=2.
```
 <Screen.RawHeader>
      <Flex mx={2}>
        <Flex>//Header</Flex>
      </Flex>
      <Separator />
    </Screen.RawHeader>
```
Is that fine?

**Pull request merged**